app/models/track.rb

```diff
@@ -29,8 +29,12 @@ class Track < ActiveRecord::Base
   validates :room, presence: true, if: :self_organized_and_accepted_or_confirmed?
   validates :relevance, presence: true, if: :self_organized?
   validates :description, presence: true, if: :self_organized?
-  validate :valid_dates
-  validate :valid_room, if: :self_organized_and_accepted_or_confirmed?
+  validate :start_date_after_conference_start_date
```

Comment:
I'd combine the test related to the conference date into 1, dates_within_conference_dates or something similar, and I would test if start_date is within the range of conf.start_date..conf.end_date

As long as  you have the validation start_date_before_end_date, that should be sufficient. What do you think?